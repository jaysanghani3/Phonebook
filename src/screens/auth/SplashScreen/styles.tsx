import {StyleSheet} from 'react-native';
import Colors from '../../../utils/Colors';

const styles = StyleSheet.create({
  phonebook: {fontSize: 50, color: Colors.appColor, fontWeight: '800'},
  logo: {width: 100, height: 100},
});

export default styles;
