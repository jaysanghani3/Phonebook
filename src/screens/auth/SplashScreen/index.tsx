import {View, Text, Image} from 'react-native';
import React, {useEffect} from 'react';
import auth from '@react-native-firebase/auth';
import globalStyles from '../../../utils/globalStyles';
import styles from './styles';
import Strings from '../../../translations/Strings';
import Icons from '../../../assets';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RouteStackParamsList} from '../../../navigators/AppNavigators';
import {Routes} from '../../../navigators/routes';

type SplashScreenProps = NativeStackScreenProps<
  RouteStackParamsList,
  Routes.SplashScreen
>;

const SplashScreen = ({navigation}: SplashScreenProps) => {
  useEffect(() => {
    setTimeout(() => {
      const unsubscribe = auth().onAuthStateChanged(user => {
        if (user === null) {
          navigation.replace('LoginScreen');
          return;
        } else if (user.emailVerified) {
          navigation.replace(Routes.HomeScreen);
        }
      });
      return unsubscribe;
    }, 2000);
  }, [navigation]);

  return (
    <View style={globalStyles.flexCenter}>
      <Image source={Icons.logo} style={styles.logo} />
      <Text style={styles.phonebook}>{Strings.phonebook}</Text>
    </View>
  );
};

export default SplashScreen;
