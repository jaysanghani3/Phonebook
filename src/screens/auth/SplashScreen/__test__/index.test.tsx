import renderer from 'react-test-renderer';
import SplashScreen from '../index';

it('renders correctly', () => {
  const tree = renderer
    .create(<SplashScreen navigation={undefined} route={undefined} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
