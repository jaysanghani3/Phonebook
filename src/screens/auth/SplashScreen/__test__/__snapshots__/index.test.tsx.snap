// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/logo.png",
      }
    }
    style={
      {
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      {
        "color": "#66347f",
        "fontSize": 50,
        "fontWeight": "800",
      }
    }
  >
    Phone
  </Text>
</View>
`;
